Sources/BeamsTokenProvider.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+@objc public class BeamsTokenProvider: NSObject, TokenProvider {
+    public let authURL: String
+    public let getAuthData: () -> AuthData
+
+    public init(authURL: String, getAuthData: @escaping () -> (AuthData)) {
+        self.authURL = authURL
+        self.getAuthData = getAuthData
+    }
+
+    func fetchToken(userId: String, completion: @escaping (_ response: String) -> Void) {
```

Comment:
I think this `completion` closure should be able to take some sort of error representation. That could be making the `response` type `String?` or it could be adding a separate parameter called `error`, or it could be calling it with a `Result` type of some variety.

As it is, there doesn't seem to be a convention for what you call the `completion` closure with if there was an error. Sometimes it's an empty string, sometimes it's `"Something went wrong..."`.

Reply:
Done in https://github.com/pusher/push-notifications-swift/pull/98/commits/d81555956f3e322ee70e7af33d420b9ee282d1a8.